src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
+
+        int newQuantity = currentQuantity - quantity;
+
+        if (newQuantity < 0) {
+            throw new RuntimeException("Resulting quantity for " + fruit + " cannot be negative.");
+        }
+
+        inventory.put(fruit, newQuantity);
```

Comment:
The operation of updating inventory is correct. However, remember to always check the resulting balance before saving it in the storage according to the checklist. In this case, the check on lines 9-11 serves that purpose, so you are good to go.